src/porepy/viz/solver_statistics.py

```diff
@@ -0,0 +1,323 @@
+            model,
+            [[grid] for grid in model.mdg.subdomains() + model.mdg.interfaces()],
+            add_grid_info=True,
+        )
+        for nd in range(self.nd + 1):
+            self._variable_data[f"subdomains_{nd}"] = (
+                pp.DiagnosticsMixin._variable_data(
+                    model,
+                    [[grid for grid in model.mdg.subdomains(dim=nd)]],
+                    add_grid_info=False,
+                )
+            )
+        for nd in range(self.nd):
+            self._variable_data[f"interfaces_{nd}"] = (
+                pp.DiagnosticsMixin._variable_data(
+                    model,
+                    [[grid for grid in model.mdg.interfaces(dim=nd)]],
+                    add_grid_info=False,
+                )
+            )
+
+        self._equations_data["dense"] = pp.DiagnosticsMixin._equations_data(
+            model,
+            [[grid for grid in model.mdg.subdomains() + model.mdg.interfaces()]],
+            add_grid_info=False,
+        )
+        self._equations_data["sparse"] = pp.DiagnosticsMixin._equations_data(
+            model,
+            [[grid] for grid in model.mdg.subdomains() + model.mdg.interfaces()],
+            add_grid_info=True,
+        )
+        for nd in range(self.nd + 1):
+            self._equations_data[f"subdomains_{nd}"] = (
+                pp.DiagnosticsMixin._equations_data(
+                    model,
+                    [[grid for grid in model.mdg.subdomains(dim=nd)]],
+                    add_grid_info=False,
+                )
+            )
+        for nd in range(self.nd):
+            self._equations_data[f"interfaces_{nd}"] = (
+                pp.DiagnosticsMixin._equations_data(
+                    model,
+                    [[grid for grid in model.mdg.interfaces(dim=nd)]],
+                    add_grid_info=False,
+                )
+            )
+
+        self.init_complete: bool = not self._equations_data["dense"] == ()
+        """Flag indicating if the initialization of the statistics object is complete."""
+
+        self.path: Optional[Path] = path
+        """Path to save the statistics object; if None, no saving is performed."""
+        self.num_iteration: int = 0
+        """Number of non-linear iterations performed for current time step."""
+        self.increment_errors: list[float] = []
+        """List of increments for each non-linear iteration."""
+        self.residual_errors: list[float] = []
+        """List of residual for each non-linear iteration."""
+        self.sub_increment_errors: dict["str", list[float]] = {}
+        """Dictionary of subincrements for each non-linear iteration."""
+        self.sub_residual_errors: dict["str", list[float]] = {}
+        """Dictionary of subresiduals for each non-linear iteration."""
+        self.init_sub_residual_errors: dict["str", float] = {}
+        """Dictionary of subinitial residuals for each time step."""
```

Comment:
A consistent choice of names is aimed at here. Yet, in my opinion, it starts becoming more cumbersome. The simplest would be to use `increments`, `residuals`, `increment_components`, `residual_components`. This would be my preferred suggestion. Consistently, in the other files.